Documentation/guides/OneDotNetEmbeddedResources.md

```diff
@@ -0,0 +1,288 @@
+a custom format that Android does not understand, we can't leave the
+files as-is. Android command-line tooling like `aapt2` works with
+files and directories on disk.
+
+Java binding projects have a slight variation:
+
+* `@(LibraryProjectZip)` can be `.aar` or `.zip` files and is
+  repackaged into `__AndroidLibraryProjects__.zip`. You can only have
+  one `@(LibraryProjectZip)` per project.
+* No support for `@(AndroidResource)` or `@(AndroidAsset)`.
+* `@(EmbeddedJar)` and `@(EmbeddedReferenceJar)` are packaged directly
+  as `EmbeddedResource` files.
+
+## New Approach in .NET 6
+
+We want to consider an approach that is NuGet-centric. When the above
+implementation was written, NuGet was just getting started. Developers
+still had the pattern of copying .NET assemblies from `bin` and
+committing it to their source control of choice. The single file
+approach worked well for this scenario.
+
+If we look at the general structure of a [NuGet package][nuget]:
+
+    lib/
+        net6.0-android29/Foo.dll
+        net6.0-android30/Foo.dll
+    # Optional reference assemblies
+    ref/
+        net6.0-android29/Foo.dll
+        net6.0-android30/Foo.dll
+    # Optional native libraries
+    runtimes/
+        android-arm/native/libFoo.so
+        android-arm64/native/libFoo.so
+        android-x86/native/libFoo.so
+        android-x64/native/libFoo.so
+
+There is not a great place where all Android file types would fit
+following this pattern.
+
+In Android Studio, Android libraries are packaged as [`.aar`][aar]
+files. A Xamarin.Android library, `Foo.csproj`, could also generate an
+[`.aar`][aar] file in its `$(OutputPath)`:
+
+    Foo.aar
+        classes.jar
+        res/
+        assets/
+        libs/*.jar
+        jni/[arch]/*.so
+
+`@(AndroidEnvironment)` files are specific to Xamarin.Android, so we
+can make one addition to the file format:
+
+    Foo.aar
+        .netenv/MyEnvironment.txt
+
+The name `.netenv` would be unlikely to collide with anything Google
+creates in the `.aar` file format in the future. It should also be
+completely ignored by Android tooling.
+
+So the output of `Foo.csproj` would look like:
+
+    bin/Debug/$(TargetFramework)/
+        Foo.dll
+        Foo.aar
+    bin/Release/$(TargetFramework)/
+        Foo.dll
+        Foo.aar
+
+If you ran the `Pack` target, you would get a `.nupkg` file with:
+
+    lib/
+        net6.0-android29/Foo.dll
+        net6.0-android29/Foo.aar
+        net6.0-android30/Foo.dll
+        net6.0-android30/Foo.aar
+
+When consuming the `.nupkg` files, Xamarin.Android will still have to
+extract [`.aar`][aar] files on disk for command-line tooling such as
+`javac`, `d8/r8` and `manifestmerger`. If users want to copy around
+loose build output and consume it, there will only be 1 additional
+[`.aar`][aar] file they will need to copy.
+
+### Java/Kotlin Dependencies
+
+A `Foo.csproj` might include `bar.aar` and `baz.jar` files that are
```

Comment:
If you're going to mention a `baz.jar` file, then the following example output should also mention it. :-)